pkg/tkctl/cmd/check/check.go

```diff
@@ -0,0 +1,460 @@
+	}
+	if container := getContainerFromSts(pump, "pump"); container != nil {
+		pumpVersion = getImageVersion(container.Image)
+	}
+	if pdVersion != "" && pdVersion == tikvVersion &&
+		pdVersion == tidbVersion {
+		if pump == nil || pdVersion == pumpVersion {
+			return true
+		}
+		if drainer == nil || pdVersion == drainerVersion {
+			return true
+		}
+	}
+	errorPrintf("cluster component version mismatch: (pd, %s), (tikv, %s), (tidb, %s), (pump, %s)\n", pdVersion, tikvVersion, tidbVersion, pumpVersion)
+	return false
+}
+
+func getContainerFromSts(sts *appsv1.StatefulSet, name string) *corev1.Container {
+	if sts == nil {
+		return nil
+	}
+	for _, container := range sts.Spec.Template.Spec.Containers {
+		if container.Name == name {
+			return &container
+		}
+	}
+	return nil
+}
+
+func getImageVersion(image string) string {
+	str := strings.Split(image, ":")
+	if len(str) != 2 && len(str) != 3 {
+		return ""
+	}
+	return str[len(str)-1]
+}
+
+func (o *CheckOptions) renderChart() (map[string]string, error) {
+	name := o.Chart
+	version := o.ChartVersion
+	var verify bool
+	var repoURL, username, password, keyring, certFile, keyFile, caFile string
+	cp, err := locateChartPath(repoURL, username, password, name, version, verify, keyring, certFile, keyFile, caFile)
+	if err != nil {
+		return nil, err
+	}
+	infoPrintln("chart path:", cp)
+
+	c, err := chartutil.Load(cp)
+	if err != nil {
+		return nil, err
+	}
+	renderer := engine.New()
+	caps := &chartutil.Capabilities{
+		APIVersions:   chartutil.DefaultVersionSet,
+		KubeVersion:   chartutil.DefaultKubeVersion,
+		TillerVersion: tversion.GetVersionProto(),
+	}
+
+	valuesFile, err := os.Open(o.ValuesFile)
+	defer valuesFile.Close()
+	if err != nil {
+		return nil, err
+	}
+	bytes, err := ioutil.ReadAll(valuesFile)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &chartproto.Config{Raw: string(bytes), Values: map[string]*chartproto.Value{}}
+	kv, err := semver.NewVersion(kubeVersion)
+	if err != nil {
+		return nil, err
+	}
+	caps.KubeVersion.Major = fmt.Sprint(kv.Major())
+	caps.KubeVersion.Minor = fmt.Sprint(kv.Minor())
+	caps.KubeVersion.GitVersion = fmt.Sprintf("v%d.%d.0", kv.Major(), kv.Minor())
+
+	options := chartutil.ReleaseOptions{
+		Name:      o.TidbClusterName,
+		Time:      timeconv.Now(),
+		Namespace: o.Namespace,
+	}
+	vals, err := chartutil.ToRenderValuesCaps(c, config, options, caps)
+	if err != nil {
+		return nil, err
+	}
+
+	return renderer.Render(c, vals)
+}
+
+func (o *CheckOptions) checkTidbService(svc corev1.Service) {
+	if svc.Spec.Type == "LoadBalancer" || svc.Spec.Type == "NodePort" {
+		policy := "Cluster"
+		if svc.Spec.ExternalTrafficPolicy == "Local" {
+			policy = "Local"
+		}
+		warnPrintf("External Traffic Policy: %s\n", policy)
+	}
+	annotations := svc.Annotations
+	if o.Cloud != "" {
+		if svc.Spec.Type != "LoadBalancer" || annotations == nil {
+			errorPrintln("TiDB Service not exposed as load balancer on", o.Cloud)
```

Comment:
Miss `%s`?

Reply:
This is `errorPrintln`, no need to add `%s`.

Reply:
For `svc.Spec.Type == "LoadBalancer" && annotations == nil` case, the error log does not make sense.

Reply:
Should return here?